forge.tf

```diff
@@ -0,0 +1,21 @@
+resource "azurerm_container_registry" "forge" {
+  name                = replace("${var.resource_group}forge", "/[^a-zA-Z0-9]/", "")
+  resource_group_name = data.azurerm_resource_group.aks.name
+  location            = data.azurerm_resource_group.aks.location
+  sku                 = var.registry_tier
+  admin_enabled       = false
+
+  // Premium only
+  public_network_access_enabled = var.registry_tier == "Premium" ? false : true
```

Comment:
Do we need to consider private endpoints or anything? (speed might be a factor given the massive base image)

Reply:
The issue is the tier. Its only supported with Premium hence the logic. If you try to enable it for Basic or Standard, TF will error out.
This is one of those things where cost comes into play.
Do we do one thing for dev and another for prod?

Speedwise I am not certain. Most of the clouds will optimize "public" traffic back to their networks anyhow. Its more whether the traffic every thinks about hitting the internet or not.

Reply:
I think we can cross that bridge when we come to it. Given this is all generally used for dev, it's not of great concern immediately.